src/test/java/kg/jarkyn/server/ServerTest.java

```diff
@@ -0,0 +1,73 @@
+package kg.jarkyn.server;
+
+import kg.jarkyn.server.Incoming.Requester;
+import kg.jarkyn.server.Outgoing.Response;
+import org.junit.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+import static org.junit.Assert.assertEquals;
+
+public class ServerTest {
+    private Server server;
+    private ResponderDouble responder;
+    private RequestingSocketDouble requestingSocket;
+    private ReceivingSocketDouble serverSocket;
+
+    @Test
+    public void receivesAndRespondsToRequest() {
+        requestingSocket = new RequestingSocketDouble();
+        serverSocket = new ReceivingSocketDouble(requestingSocket);
+        responder = new ResponderDouble();
+        server = new Server(serverSocket, responder);
+
+        server.run();
+
+        ByteArrayOutputStream outputStream = (ByteArrayOutputStream) requestingSocket.getOutputStream();
+        assertEquals("hello", new String(outputStream.toByteArray()));
+    }
+
+    private class ReceivingSocketDouble extends ReceivingSocket {
+        private RequestingSocketDouble requestingSocket;
+
+        public ReceivingSocketDouble(RequestingSocketDouble requestingSocket) {
+            super(null);
+            this.requestingSocket = requestingSocket;
+        }
+
+        public RequestingSocketDouble accept() {
+            return requestingSocket;
+        }
+    }
+
+    private class ResponderDouble implements Outgoing.Responder {
+        public Response prepareResponse(Requester requester) {
+            return new Response("hello");
+        }
+
+        @Override
+        public void sendResponse(Requester requester, Response response) {
+            try {
+                requester.getOutputStream().write(response.getBytes());
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    private class RequestingSocketDouble implements Requester {
+        private OutputStream outputStream = new ByteArrayOutputStream();
+
+        @Override
+        public InputStream getInputStream() {
+            return null;
+        }
+
+        public OutputStream getOutputStream() {
+            return outputStream;
```

Comment:
It's interesting to see that you have a test that verifies the `getOutputStream()` of a `Requester`, yet in the [production code this method returns `null`](https://github.com/indykid/cobSpecServer/pull/2/files#diff-d5221bed05ee163e64b2378cdf15ea17R31).
There only `getInputStream()` is used and implemented, but that's not exercised in a test here.

I'd be interested to see a `main()` with the existing code on how it is used right now.